src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java

```diff
@@ -123,8 +126,21 @@ private JSONBuilder writeGeomCollection(GeometryCollection collection) {
         return this.endArray();
     }
 
-    private JSONBuilder writeCoordinates(Coordinate[] coords) throws JSONException {
-        return writeCoordinates(new CoordinateArraySequence(coords));
+    private JSONBuilder writeCoordinate(Point point) throws JSONException {
+        CoordinateSequence seq = point.getCoordinateSequence();
+        if (!seq.hasM() || !encodeMeasures) {
```

Comment:
logic like this makes me wish for an EnumSet of XYZ, XYZM.

Reply:
I see your point I will rewrite a more generic method and simplify this syntax.